decidim-core/app/views/decidim/shared/_flag_modal.html.erb

```diff
@@ -0,0 +1,19 @@
+<div class="reveal" id="flagModal" data-reveal>
+  <div class="reveal__header">
+    <h3 class="reveal__title"><%= t('.title') %></h3>
+    <button class="close-button" data-close aria-label="<%= t('.close') %>" type="button">
+      <span aria-hidden="true">&times;</span>
+    </button>
+  </div>
+  <% if resource.reported_by? current_user %>
```

Comment:
This whole bunch of logic can only be done when the user is logged in. We should either allow anonymous users to report or tell the user she can't report unless she logs in.